scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java

```diff
@@ -285,162 +318,241 @@ protected void logDataspacesStatus(String message, DataspacesStatusLevel level)
-                                    .copyFrom(
-                                            finaldsfo,
-                                            FileSelector.SELECT_SELF);
-
-                            if (!finalOut.exists()) {
-                                logger.error("There was a problem during the copy of " + finaldsfo.getRealURI() + " to " + finalOut.getRealURI() +
-                                        "/" + finalRelativePath + ". File not present after copy.");
-                                logDataspacesStatus("There was a problem during the copy of " + finaldsfo.getRealURI() + " to " + finalOut.getRealURI() +
-                                                "/" + finalRelativePath + ". File not present after copy.",
-                                        DataspacesStatusLevel.ERROR);
-                            }
-                            return true;
-                        }
-                    }));
+    /*
+     * Create the folder hierarchy and select the files to copy
+     * from the specified list of FileObjects.
+     */
+    private void createFolderHierarchySequentially(
+            DataSpacesFileObject destination, String spaceUri, List<DataSpacesFileObject> spaceFiles,
+            Map<String, DataSpacesFileObject> filesToCopy) throws FileSystemException {
 
-                }
-                relPathes.add(relativePath);
-            }
+        for (DataSpacesFileObject fileObject : spaceFiles) {
+            String relativePath = relativize(spaceUri, fileObject);
 
-            StringBuilder exceptionMsg = new StringBuilder();
-            String nl = System.lineSeparator();
-            for (Future<Boolean> f : transferFutures) {
-                try {
-                    f.get();
-                } catch (InterruptedException | ExecutionException e) {
-                    logger.error("", e);
-                    exceptionMsg.append(StackTraceUtil.getStackTrace(e)).append(nl);
-                }
+            DataSpacesFileObject target = destination.resolveFile(relativePath);
+
+            if (target.isFolder()) {
+                target.createFolder();
+            } else if (target.isFile()) {
+                target.getParent().createFolder();
             }
-            if (exceptionMsg.length() > 0) {
-                toBeThrown = new FileSystemException(
-                        "Exception(s) occurred when transferring input files : " + nl + exceptionMsg.toString());
+
+            DataSpacesFileObject oldFileObject = filesToCopy.put(relativePath, fileObject);
+            if (oldFileObject != null) {
+                String message = fileObject.getRealURI()
+                        + " will be copied instead of " + oldFileObject.getRealURI() + ".\n "
+                        + "Precedence order is output space, input space, user space, global space.";
+                logger.warn(message);
+                logDataspacesStatus(message, DataspacesStatusLevel.WARNING);
+            };
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this empty statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)